frontend/packages/console-app/src/components/detect-perspective/__tests__/useValuesForPerspectiveContext.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { useExtensions } from '@console/plugin-sdk';
+import { useUserSettingsCompatibility } from '@console/shared';
+import {
+  PREFERRED_PERSPECTIVE_USER_SETTING_VALUE_LATEST,
+  usePreferredPerspective,
+} from '../../user-preferences';
+import { useValuesForPerspectiveContext } from '../useValuesForPerspectiveContext';
+import { mockPerspectiveExtensions } from './perspective.data';
+
+jest.mock('@console/plugin-sdk/src/api/useExtensions', () => ({
+  useExtensions: jest.fn(),
+}));
+
+jest.mock('@console/shared/src/hooks/useUserSettingsCompatibility', () => ({
+  useUserSettingsCompatibility: jest.fn(),
+}));
+
+jest.mock('../../user-preferences/perspective/usePreferredPerspective', () => ({
+  usePreferredPerspective: jest.fn(),
+}));
+
+const useExtensionsMock = useExtensions as jest.Mock;
+const useUserSettingsCompatibilityMock = useUserSettingsCompatibility as jest.Mock;
+const usePreferredPerspectiveMock = usePreferredPerspective as jest.Mock;
+
+describe('useValuesForPerspectiveContext', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should return undefined for PerspectiveType if the perspective from user settings does not match any of the perspectives received from extensions', () => {
+    useExtensionsMock.mockReturnValue(mockPerspectiveExtensions);
+    useUserSettingsCompatibilityMock.mockReturnValue(['acm', () => {}, true]);
+    usePreferredPerspectiveMock.mockReturnValue(['latest', true]);
```

Comment:
Did you want return latest here? It returns undefined anyway because acm perspective is not defined, right?

```suggestion
    usePreferredPerspectiveMock.mockReturnValue([PREFERRED_PERSPECTIVE_USER_SETTING_VALUE_LATEST, true]);
```

Reply:
missed this. Thanks!